File: elixir/bob/bob.exs
defmodule Bob do

  def isShouting(input) do
    String.upcase(input) == input and String.downcase(input) != input
  end

  def hey(input) do

    cond do
      String.ends_with? input, "?" -> "Sure."
      isShouting(input) -> "Whoa, chill out!"
      String.trim(input) == "" -> "Fine. Be that way!"
      true -> "Whatever."
    end
  end
end

File: haskell/bob/src/Bob.hs
module Bob (responseFor) where

import Data.Char (isLower, isSpace, isUpper)

responseFor :: String -> String
responseFor string = 
  | all isSpace string = "Fine. Be that way!"
  | any isUpper string && not (any isLower string) = "Whoa, chill out!"
  | last string == '?' = "Sure."
  | otherwise = "Whatever."